### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by mraafaykhan**
refactored models for screens and prototypes

### CODE REVIEW

File: api/models/prototype.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const mongoose = require('mongoose');
+const Schema = require('mongoose').Schema;
+
+const PrototypeSchema = mongoose.Schema({
+  _id: { type: Schema.Types.ObjectId , auto:true},  
+  user : {type: Schema.Types.ObjectId},
+  name: { type: String },
+  created_at: { type: Date, default: Date.now },
+  screens:[{
+    _id: {type: Schema.Types.ObjectId, default: []}
```
Comment (hamxabaig) [inline api/models/prototype.js]: You need to add a key here with ref to screens model. `ref: 'screens'`

File: api/models/screens.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+const Schema = require('mongoose').Schema;
+
+const ScreensModel = mongoose.Schema(
+  {
+
+    _id: { type: Schema.Types.ObjectId, auto:true },
+    // Double keying, prototype model is saving keys for screens and screens for prototypes that it belongs to 
+    // but we may need it in some case
+    // making life easier, optimize later
+    prototype: { type: Schema.Types.ObjectId },
```
Comment (hamxabaig) [inline api/models/screens.js]: add ref here `ref: 'prototypes'`

File: api/routes/prototypeRoutes.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      path: "/prototypes",
+
+      config: {
+        validate:
+          {
+            payload: {
+              name: Joi.string().required()
+            }
+          },
+        auth: false,
```
Comment (hamxabaig) [inline api/routes/prototypeRoutes.js]: We need auth here, `auth: 'jwt'`

File: api/controllers/prototypeController.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+let Boom = require("boom");
+
+const prototypeController = {};
+
+
+prototypeController.createPrototype = function (req, res) {
+
+  let protoype = new Prototype({
+    name: req.payload.name || "prototype1"
+    // todo replace with something meaningful if user doesnt provide a name for prototype
```
Comment (hamxabaig) [inline api/controllers/prototypeController.js]: not needed, its necessary

File: api/routes/prototypeRoutes.js (hunk 3 above)
Comment (mraafaykhan) [reply api/routes/prototypeRoutes.js]: i've added this in the read.me for dev lets keep it to false.

File: api/controllers/prototypeController.js (hunk 4 above)
Comment (mraafaykhan) [reply api/controllers/prototypeController.js]: you will have to do it at someplace, 
for example creating a new file in pages or keynote, if you leave it blank it'll say newDoc1 or pages1
would be good for ux i think.

File: api/routes/prototypeRoutes.js (hunk 3 above)
Comment (hamxabaig) [reply api/routes/prototypeRoutes.js]: @mraafaykhan I'll be testing it in app, so thats why.

File: api/models/screens.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+const Schema = require('mongoose').Schema;
+
+const ScreensModel = mongoose.Schema(
+  {
+
+    _id: { type: Schema.Types.ObjectId, auto:true },
+    // Double keying, prototype model is saving keys for screens and screens for prototypes that it belongs to 
+    // but we may need it in some case
+    // making life easier, optimize later
+    createdBy: { type: Schema.Types.ObjectId , ref:"prototype"},
```
Comment (hamxabaig) [inline api/models/screens.js]: @mraafaykhan you had it right before. It should be `prototype: { type: Schema.Types.ObjectId , ref:"prototypes"},`

File: api/models/prototype.js [hunk 6]
```diff
@@ -2,31 +2,12 @@ const mongoose = require('mongoose');
...
-        text: {
-          type: String
-        },
-        type: {
-          type: String
-        }
-      }
-    ]
+  screens:[{
+    _id: {type: Schema.Types.ObjectId, default: []}
```
Comment (hamxabaig) [inline api/models/prototype.js]: add ref here

File: api/models/prototype.js [hunk 7]
```diff
@@ -2,31 +2,12 @@ const mongoose = require('mongoose');
 const Schema = require('mongoose').Schema;
 
 const PrototypeSchema = mongoose.Schema({
+  _id: { type: Schema.Types.ObjectId , auto:true},  
+  user : {type: Schema.Types.ObjectId},
```
Comment (hamxabaig) [inline api/models/prototype.js]: ref here

**Comment by mraafaykhan** [comment]
review now please

File: api/models/prototype.js [hunk 8]
```diff
@@ -2,31 +2,12 @@ const mongoose = require('mongoose');
 const Schema = require('mongoose').Schema;
 
 const PrototypeSchema = mongoose.Schema({
+  _id: { type: Schema.Types.ObjectId , auto:true},  
+  user : {type: Schema.Types.ObjectId , ref :"User"},
```
Comment (hamxabaig) [inline api/models/prototype.js]: ref needs to be users

File: api/models/screens.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+const Schema = require('mongoose').Schema;
+
+const ScreensModel = mongoose.Schema(
+  {
+
+    _id: { type: Schema.Types.ObjectId, auto:true },
+    // Double keying, prototype model is saving keys for screens and screens for prototypes that it belongs to 
+    // but we may need it in some case
+    // making life easier, optimize later
+    protoype: { type: Schema.Types.ObjectId , ref:"prototype"},
```
Comment (hamxabaig) [inline api/models/screens.js]: ref needs to be prototypes.

File: api/models/screens.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+      {
+        dimensions: {
+          x: { type: Number },
+          y: { type: Number },
+          height: { type: Number },
+          width: { type: Number },
+          color: { type: String }
+        },
+        link: {
+          type: Schema.Types.ObjectId
```
Comment (hamxabaig) [inline api/models/screens.js]: ref needs to be screens.

**Pull request merged**